src/modules/job-exec/bulk-exec.c

```diff
@@ -477,6 +477,7 @@ int bulk_exec_push_cmd (struct bulk_exec *exec,
 
     if (zlist_append (exec->commands, c) < 0) {
         exec_cmd_destroy (c);
+        errno = ENOMEM;
```

Comment:
not to nit pick, but is ENOMEM impossible from zlist_append()? b/c of their internal asserts? No worries though, this is everywhere in flux-core, we gotta assign some ERRNO for the time being.

Reply:
zlist(3) says it can return -1 on out of memory:
```c
  //  Append an item to the end of the list, return 0 if OK or -1 if this
  //  failed for some reason (out of memory). Note that if a duplicator has
  //  been set, this method will also duplicate the item.
  CZMQ_EXPORT int
      zlist_append (zlist_t *self, void *item);
```
I didn't check the source but coding to the docs seemed like the right thing to do.

Reply:
ahhh i forgot that my comment tweak was in a PR

https://github.com/zeromq/czmq/pull/2266

no biggie, we can leave for now

Reply:
We do have other places in the code like [here](https://github.com/flux-framework/flux-core/blob/master/src/modules/resource/reslog.c#L154) and [here](https://github.com/flux-framework/flux-core/blob/master/src/modules/kvs/lookup.c#L133) that set `errno` to ENOMEM after a `zlist_append()` failure.  So maybe a future cleanup should address this?

Reply:
yeah, a future cleanup.